akka-sample-sharding-scala/src/main/scala/sample/sharding/TemperatureService.scala

```diff
@@ -0,0 +1,64 @@
+package sample.sharding
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+import akka.actor.typed.{ ActorRef, Behavior }
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.cluster.sharding.typed.scaladsl.{ ClusterSharding, Entity, EntityTypeKey }
+
+object TemperatureService {
+
+  trait TemperatureEvent
+  // Update a random device
+  case object UpdateDevice extends TemperatureEvent
+  case object ReadTemperatures extends TemperatureEvent
```

Comment:
Looks more like commands than events to me as they are imperative, either change to `Command` top level or change these to event style naming `DeviceUpdateTick` or something like that.  I'd vote for command. Also drop the `Temperature` prefix, that's already in the scope.

Reply:
yes but because this also receives `Temperature` which is not a command, a more general type is needed.

Reply:
I don't think just Event makes sense.

Reply:
It is/was a fabricated scenario so the sample would need an overhaul to be correct and I don't think that was the goal of this update to typed exercise. Devices reading temperatures would be distributed like weather stations reporting to the cluster of services receiving readings.

Reply:
In the training materials we solved the command vs event by stating all messages as imperative, but using a top level type `Message` I think works fine as well, then a message can either be an indivudual command or an event.

Just `Event`/`Command`/`Message` makes sense because it is not referred to like that elsewhere, and there is no risk for confusion. Whenever referenced from another actor you should use the actor object as a prefix anyway, so that would always be `TemperatureService.Message`.